tf-aws/modules/iam/main.tf

```
@@ -119,4 +119,170 @@ resource "aws_iam_role_policy" "process_monetary_transactions_lambda_policy" {
 resource "aws_iam_role_policy_attachment" "AWSLambdaVPCAccessExecutionRole" {
   role       = aws_iam_role.process_monetary_transactions_lambda_role.name
   policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole"
-}
\ No newline at end of file
+}
+
+# EKS RELATED ROLES
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "eks" {
+  name = "eks-cluster-prod"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "eks.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "prod-AmazonEKSClusterPolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
+  role       = aws_iam_role.eks.name
+}
+
+resource "aws_iam_role" "nodes" {
+  name = "eks-node-group-nodes"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "nodes-AmazonEKSWorkerNodePolicy" {
+  # Grant access to EC2 and EKS
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+  role       = aws_iam_role.nodes.name
+}
+
+resource "aws_iam_role_policy_attachment" "nodes-AmazonEKS_CNI_Policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+  role       = aws_iam_role.nodes.name
+}
+
+resource "aws_iam_role_policy_attachment" "nodes-AmazonEC2ContainerRegistryReadOnly" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+  role       = aws_iam_role.nodes.name
+}
+
+resource "aws_iam_user" "dev" {
+  name = "dev"
+}
+
+resource "aws_iam_policy" "eks_read_only" {
+  name = "AmazonEKSDeveloperPolicy"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "eks:DescribeCluster",
+          "eks:ListClusters"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_user_policy_attachment" "dev-eksReadOnly" {
+  user       = aws_iam_user.dev.name
+  policy_arn = aws_iam_policy.eks_read_only.arn
+}
+
+resource "awscc_eks_access_entry" "dev" {
+  cluster_name      = var.eks_cluster_name
+  principal_arn     = aws_iam_user.dev.arn
+  kubernetes_groups = ["eks-dev"]
+}
+
+resource "aws_iam_user" "admin" {
+  name = "admin"
+}
+
+resource "aws_iam_role" "eks_admin" {
+  name = "eks-node-group-nodes"
```

```suggestion
  name = "eks-admin-role"
```
on top have another IAM with the same name, not sure if aws allow

---

oops i resolved